pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -1003,13 +1016,46 @@ func (r *Reconciler) runReconcileSeedFlow(
 				Name: "Deploying ETCD Druid",
 				Fn:   etcdDruid.Deploy,
 			})
+			_ = g.Add(flow.Task{
+				Name: "Deploying Fluent Operator Resources and wait until they become ready",
+				Fn:   component.OpWait(fluentOperatorResources).Deploy,
+			})
+			_ = g.Add(flow.Task{
+				Name: "Deploying Fluent Operator",
+				Fn:   component.OpWait(fluentOperator).Deploy,
+			})
 		)
 	}
 
 	if err := g.Compile().Run(ctx, flow.Opts{Log: log}); err != nil {
 		return flow.Errors(err)
 	}
 
+	if loggingEnabled {
+		// TODO(Kristian-ZH): Delete this in a future when the operator releases a version with this improvement: Ref https://github.com/fluent/fluent-operator/issues/654
+		// Waiting for fluent-bit service to be created
+		fluentBitService := &corev1.Service{}
+		if err := retry.UntilTimeout(ctx, 5*time.Second, time.Minute, func(ctx context.Context) (done bool, err error) {
+			if err = seedClient.Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, v1beta1constants.DaemonsetNameFluentBit), fluentBitService); err != nil {
+				return retry.MinorError(fmt.Errorf("fluent-bit service does not exist yet"))
+			}
+			return retry.Ok()
+		}); err != nil {
+			return err
+		}
+
+		// Patching fluent-bit service with the network policy annotations
+		patch = client.MergeFrom(fluentBitService.DeepCopy())
+		fluentBitService.ObjectMeta.Annotations = map[string]string{
+			v1beta1constants.AnnotationNetworkPolicyFromPolicyPodLabelSelector: v1beta1constants.NetworkPolicyAllSeedScrapeTargets,
+			v1beta1constants.AnnotationNetworkPolicyFromPolicyAllowedPorts:     `[{"port":"2020","protocol":"TCP"},{"port":"2021","protocol":"TCP"}]`,
+		}
```

Comment:
We are waiting for a release of the fluent-operator this week where it will be possible to add the annotation through the CR and I will completely remove this patch soon